widgets/src/lib/InferenceWidget/widgets/StructuredDataClassificationWidget/StructuredDataClassificationWidget.svelte

```diff
@@ -0,0 +1,212 @@
+<script>
+	import type {
+		WidgetProps,
+		TableData,
+		HighlightCoordinates,
+	} from "../../shared/types";
+
+	import { onMount } from "svelte";
+	import WidgetTableInput from "../../shared/WidgetTableInput/WidgetTableInput.svelte";
+	import WidgetSubmitBtn from "../../shared/WidgetSubmitBtn/WidgetSubmitBtn.svelte";
+	import WidgetWrapper from "../../shared/WidgetWrapper/WidgetWrapper.svelte";
+	import {
+		mod,
+		convertTableToData,
+		convertDataToTable,
+		parseJSON,
+	} from "../../shared/ViewUtils";
+	import {
+		getDemoInputs,
+		getResponse,
+		getSearchParams,
+		updateUrl,
+	} from "../../shared/helpers";
+
+	export let apiToken: WidgetProps["apiToken"];
+	export let apiUrl: WidgetProps["apiUrl"];
+	export let callApiOnMount: WidgetProps["callApiOnMount"];
+	export let model: WidgetProps["model"];
+	export let noTitle: WidgetProps["noTitle"];
+	export let shouldUpdateUrl: WidgetProps["shouldUpdateUrl"];
+
+	const columns: string[] = Object.keys(
+		model?.widgetData?.[0]?.structuredData ?? {}
+	);
+
+	let computeTime = "";
+	let error: string = "";
+	let isLoading = false;
+	let modelLoading = {
+		isLoading: false,
+		estimatedTime: 0,
+	};
+	let output: string[] = [];
+	let outputJson: string;
+	let table: string[][] = [columns];
+
+	let highlighted: HighlightCoordinates = {};
+	let highlightError = "";
+	let scrollTableToRight: () => Promise<void>;
+	let tableWithOutput: string[][];
+	$: {
+		const strucuredData = convertTableToData(table);
+		if (output?.length) {
+			strucuredData.Prediction = output;
+			const lastColIndex = Object.keys(strucuredData).length - 1;
+			highlighted = highlightOutput(output, lastColIndex);
+			scrollTableToRight();
+		} else {
+			delete strucuredData.Prediction;
+			highlighted = {};
+			if (highlightError) {
+				highlighted[highlightError] =
+					"bg-red-100 border-red-100 dark:bg-red-800 dark:border-red-800";
+				highlightError = "";
+			}
+		}
+		tableWithOutput = convertDataToTable(strucuredData);
+	}
+
+	const COLORS = ["blue", "green", "yellow", "purple", "red"] as const;
```

Comment:
@gary149 here is the list of `colors` for the Predictions. Please let me know if you have suggestions on the list.
More details:
```
let arr_unique = unique([1,1,2,2,3,3]) --> [1,2,3]
If arr_unique.length <= colors.length:
    assign color to every unique class --> all 1s are same blue, all 2s are green, etc.
else:
  assign color by iterating colors --> first element blue, second element green, nth element colors[mod(n, color.length)]
```